ui/perfherder/alerts/Pagination.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { Row, Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+
+class Paging extends React.Component {
+  constructor(props) {
+    super(props);
+    this.validated = this.props.validated;
+    this.state = {
+      page: this.validated.page ? parseInt(this.validated.page, 10) : 1,
+      count: 0,
+    };
+  }
+
+  navigatePage = page => {
+    this.setState({ page }, this.fetchAlertSummaries);
+    this.props.validated.updateParams({ page });
```

Comment:
Not every view has access to `this.props.validated` so to make this component more versatile and reusable, you should pass in `updateParams' and `page` props in to this component, rather than extracting those values from `this.props.validated`.

Also, you do not have access to `this.fetchAlertSummaries` since that method is not a member of this class (running the `yarn lint --fix` command frequently will catch mistakes like this). You will also need to pass that in - but again - make it generic. So in this component, your callback to setState could be `fetchData`.